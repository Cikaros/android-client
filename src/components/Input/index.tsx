import React from 'react';
import {
  StyleSheet,
  Text,
  Image,
  ViewStyle,
  View,
  TextInput,
  TouchableOpacity,
} from 'react-native';
import {NativeSyntheticEvent} from 'react-native/Libraries/Types/CoreEventTypes';
import {TextInputChangeEventData} from 'react-native/Libraries/Components/TextInput/TextInput';
import {colors} from '../../theme';

interface Props {
  onChange?: (e: NativeSyntheticEvent<TextInputChangeEventData>) => void;
  onChangeText?: (text: string) => void;
  onTextInput?: () => void;
  styles?: StyleSheet.NamedStyles<any>;
  stylesMask?: any;
  editable?: boolean;
  value?: string;
  placeHolder?: string;
  keyboardType?: string;
  error?: boolean;
  errorMessage?: string;
  clearVisible?: boolean;
  onClear?: () => void;
}

const stylesPallet = {
  NbMaskTextInput: {
    fontSize: 14,
    lineHeight: 22,
    height: 40,
    borderWidth: 1,
    borderColor: 'rgba(217, 217, 217, 1)',
    paddingHorizontal: 0,
    paddingVertical: 2,
    backgroundColor: 'white',
    flexDirection: 'row',
    alignItems: 'center',
    overFlow: 'hidden',
  } as ViewStyle,
  NbTextInput: {
    color: 'rgba(77, 77, 77, 1)',
    flex: 1,
    height: 40,
    paddingHorizontal: 8,
    paddingVertical: 2,
    fontFamily: 'Roboto',
    fontSize: 14,
    lineHeight: 22,
    fontWeight: '500',
  } as ViewStyle,

  error: {
    borderColor: colors.error,
    padding: 3,
    position: 'relative', 
    color: '#A8071A'
  } as ViewStyle,
};

const Input: React.FC<Props> = ({
  onChange,
  onChangeText,
  onTextInput,
  styles,
  stylesMask,
  editable,
  value,
  placeHolder,
  error,
  errorMessage,
  clearVisible,
  onClear,
}) => {

  return (
    <>
      <View style={[stylesPallet.NbMaskTextInput, stylesMask]}>
        <TextInput
          onChange={onChange}
          onChangeText={onChangeText}
          onTextInput={onTextInput}
          style={[
            styles,
            stylesPallet.NbTextInput
          ]}
          placeholderTextColor={'rgba(0,0,0,0.25)'}
          editable={editable}
          value={value}
          placeholder={placeHolder}
        />
        {clearVisible && (
          <TouchableOpacity
            style={{padding: 10}}
            onPress={() => onClear && onClear()}>
            <Image
              source={require('@assets/images/close-slider-4x.png')}
              style={{width: 15, height: 15, resizeMode: 'contain'}}
            />
          </TouchableOpacity>
        )}
      </View>
      {error && (
        <Text style={stylesPallet.error}>
          {errorMessage}
        </Text>
      )}
    </>
  );
};

export default React.memo(Input);
