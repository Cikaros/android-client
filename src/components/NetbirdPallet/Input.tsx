import styled from 'styled-components';
import { Text, TextInput, TouchableOpacity, View } from 'react-native';

export const NbMaskTextInput = styled(View)`
  font-size: 14px;
  padding: 0px 8px;
  border: solid 1px rgba(217, 217, 217, 1);
  background-color: white;
  flex-direction: row;
  align-items: center;
  overflow: hidden;
`;

export const NbTextInput = styled(TextInput)`
  flex: 1;
  color: rgba(77, 77, 77, 1);
  height: 32px;
  padding: 0;
  font-family: 'Roboto';
  line-height: 24px;
  font-weight: 500;
  color: rgba(0, 0, 0, 0.35);
`;

export const NbClearInput = styled(TouchableOpacity)`
  padding: 3px 0 3px 3px;
`;

export const NbLabelError = styled(Text)`
  padding-top: 5px;
  margin-left: 8px;
  font-size: 12px;
`;
