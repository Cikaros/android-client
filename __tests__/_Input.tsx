import React from 'react';
import {shallow} from 'enzyme';
import Input from '../src/components/Input/index';

interface Props {
  onChange?: jest.Mock;
  onChangeText?: jest.Mock;
  onTextInput?: jest.Mock;
  styles?: any; 
  stylesMask?: any;
  editable?: boolean;
  value?: string;
  placeHolder?: string;
  keyboardType?: string;
  error?: boolean;
  errorMessage?: string;
  clearVisible?: boolean;
  onClear?: jest.Mock;
}

describe('Input component', () => {
  const defaultProps: Props = {
    onChange: jest.fn(),
    onChangeText: jest.fn(),
    onTextInput: jest.fn(),
    styles: undefined,
    stylesMask: undefined,
    editable: true,
    value: '',
    placeHolder: 'Enter text',
    keyboardType: 'default',
    error: false,
    errorMessage: '',
    clearVisible: false,
    onClear: jest.fn(),
  };

  it('should render correctly with default props', () => {
    const wrapper = shallow(<Input {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

});
