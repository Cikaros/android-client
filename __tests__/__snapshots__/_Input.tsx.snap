// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render correctly with default props 1`] = `
<Fragment>
  <Component
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "rgba(217, 217, 217, 1)",
          "borderWidth": 1,
          "flexDirection": "row",
          "fontSize": 14,
          "height": 40,
          "lineHeight": 22,
          "overFlow": "hidden",
          "paddingHorizontal": 0,
          "paddingVertical": 2,
        },
        undefined,
      ]
    }
  >
    <Component
      editable={true}
      onChange={[MockFunction]}
      onChangeText={[MockFunction]}
      onTextInput={[MockFunction]}
      placeholder="Enter text"
      placeholderTextColor="rgba(0,0,0,0.25)"
      style={
        [
          undefined,
          {
            "color": "rgba(77, 77, 77, 1)",
            "flex": 1,
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontWeight": "500",
            "height": 40,
            "lineHeight": 22,
            "paddingHorizontal": 8,
            "paddingVertical": 2,
          },
        ]
      }
      value=""
    />
  </Component>
</Fragment>
`;
